libdnf/hy-iutil-private.hpp

```diff
@@ -41,6 +41,7 @@ char *abspath(const char *path);
 int is_readable_rpm(const char *fn);
 int mkcachedir(char *path);
 gboolean mv(const char *old_path, const char *new_path, GError **error);
+gboolean dnf_remove_recursive_v2(const gchar *path, GError **error);
```

Comment:
Hmm, why a new function, can't you extend the original one to also work for files?

Reply:
If the original one needs to be preserved, more descriptive name would be nice.

Reply:
@m-blaha actually, `_v2` etc. suffix is sometimes used when a function is updated in a non-compatible way but still mostly does the same thing. A name along the lines of `dnf_remove_recursive_file` will only confuse readers and a properly descriptive name like `dnf_remove_recursive_with_support_for_files` is too long. So in this case I'd be fine with `_v2` myself (if the original one can't be extended).

Unless you have a great name suggestion of course :grin:

Reply:
Well, I don't have one ;)

Reply:
@jrohel do we need to preserve the original version of dnf_remove_recursive?

Reply:
The original function is public one, therefore extension of functionality could be a problem (never know who and how it is used).
Also the requirement for the patch was:
1. Small
2. Compatible
3. No regressions
4. API unchanged

I support to create the new private function with extended functionality (Like it is in PR). The function could be replaced by c++17 alternatives when it will be possible.

Reply:
OK, I missed that original function was public. Merging.